vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -220,6 +220,50 @@
     delay: 20
     register: ready_hosts
     until: ready_hosts.stdout == "1"
+    
+    #### m3dt template reference check
+  - name: Get the m3dt,edit and apply a new one 
+    shell: kubectl get m3dt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers-template  -ojson | 
+              jq '.metadata.name="test-new-m3dt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' |
+              jq '.|del(.status)' | 
+              jq '.spec.templateReference="{{ CLUSTER_NAME }}-workers-template"' |
+              kubectl apply -f- 
+
+  - name: Get the m3mt,edit and apply a new one 
+    shell: kubectl get m3mt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers -ojson |
+              jq '.spec.template.spec.dataTemplate.name="test-new-m3dt"' | 
+              jq '.metadata.name="test-new-m3mt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' |
+              jq '.|del(.status)' | 
+              kubectl apply -f-
+
+  - name: Edit MachineDeployment to point to the new m3mt
+    shell: kubectl get md -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}" -ojson | 
+              jq '.spec.template.spec.infrastructureRef.name="test-new-m3mt"' | 
+              jq '.spec.strategy.rollingUpdate.maxUnavailable=1' |
+              kubectl apply -f-
+    
+  - pause:
+     minutes: 1
+
+  - name: Wait until one BMH is in ready state
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
+
+  - name: Check if one Metal3Data refers to the old template
+    shell: kubectl get m3d -n metal3 -ojson | 
+              jq -r '[ .items[] | select (.spec.templateReference=="test1-workers-template")|
```

Comment:
```suggestion
              jq -r '[ .items[] | select (.spec.templateReference=="{{ CLUSTER_NAME }}-workers-template")|
```

Reply:
Thanks, since it is a cosmetic change. I ll do it in some future PR since passing the feature test now is a big issue.

Reply:
Done